dbt_platform_helper/commands/conduit.py

```diff
@@ -207,15 +209,19 @@ def create_addon_client_task(
     elif addon_type == "redis" or addon_type == "opensearch":
         secret_name += "_ENDPOINT"
 
+    role_name = f"{addon_name}-{app.name}-{env}-conduitEcsTask"
+
     try:
-        session.client("iam").get_role(
-            RoleName=f"{app.name}-{addon_type}-{app.name}-{env}-conduitEcsTask"
-        )
-        execution_role = (
-            f"--execution-role {app.name}-{addon_type}-{app.name}-{env}-conduitEcsTask "
-        )
-    except:
+        session.client("iam").get_role(RoleName=role_name)
+        execution_role = f"--execution-role {role_name} "
+    except ClientError as ex:
         execution_role = ""
+        # We cannot catch NoSuchEntityException directly as it is a dynamically generated exception by the boto3
```

Comment:
Would something like `client.exceptions.NoSuchEntityException` not work here?

Reply:
Will check.